File: 6th Semester/Programming Paradigms/NP2 Exercises/ex5.py
# – Execute o programa no qual o usuário entre com a idade do carro e caso o valor seja menor ou igual a
# 3 anos imprima “Seu carro é novo”, caso contrario “Seu carro é velho”

def ler_inteiro(mensagem: str) -> int:
    while True:
        output = str(input(mensagem)).strip()
        if (output.replace('-','').isnumeric()):
            return int(output)
        print('Por favor insira um número inteiro válido.')

def main():
    print("Programa classificação da idade do seu carro")
    idade = ler_inteiro('Digite a idade do seu carro: ')
    if idade <= 3:
        print("Seu carro é novo")
    else:
        print("Seu carro é velho")

if __name__ == "__main__":
    main()

File: 5th Semester/Programming Paradigms/NP2 Exercises/ex5.c
//  – Execute o programa no qual o usuário entre com a idade do carro e caso o valor seja menor ou igual a
// 3 anos imprima “Seu carro é novo”, caso contrario “Seu carro é velho”
#include <stdio.h>
#include <stdlib.h>

#define TAMANHO_OUTPUT 100

int lerInteiro(char mensagem[]) {
    char output[TAMANHO_OUTPUT];
    int inteiro;
    while(1) {
        printf("%s", mensagem);
        fgets(output, TAMANHO_OUTPUT, stdin);
        if (!((inteiro = atoi(output)) == 0 && output[0] != '0'))
            return inteiro;
        printf("Por favor insira um número inteiro válido!\n");
    }
}

int main() {
    printf(lerInteiro("Digite a idade do seu carro: ") > 3 ? "Seu carro é velho\n" : "Seu carro é novo\n");
    return 0;
}